src/lbryio.js

```diff
@@ -0,0 +1,142 @@
+  if (!Lbryio.enabled) {
+    return Promise.reject(new Error(__('LBRY internal API is disabled')));
+  }
+
+  if (!(method === 'get' || method === 'post')) {
+    return Promise.reject(new Error(__('Invalid method')));
+  }
+
+  function checkAndParse(response) {
+    if (response.status >= 200 && response.status < 300) {
+      return response.json();
+    }
+    return response.json().then(json => {
+      let error;
+      if (json.error) {
+        error = new Error(json.error);
+      } else {
+        error = new Error('Unknown API error signature');
+      }
+      error.response = response; // This is primarily a hack used in actions/user.js
+      return Promise.reject(error);
+    });
+  }
+
+  function makeRequest(url, options) {
+    return fetch(url, options).then(checkAndParse);
+  }
+
+  return Lbryio.getAuthToken().then(token => {
+    const fullParams = { auth_token: token, ...params };
+    const qs = querystring.stringify(fullParams);
+    let url = `${CONNECTION_STRING}${resource}/${action}?${qs}`;
+
+    let options = {
+      method: 'GET',
+    };
+
+    if (method === 'post') {
+      options = {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/x-www-form-urlencoded',
+        },
+        body: qs,
+      };
+      url = `${CONNECTION_STRING}${resource}/${action}`;
+    }
+
+    return makeRequest(url, options).then(response => response.data);
+  });
+};
+
+Lbryio.authToken = null;
+
+Lbryio.getAuthToken = () =>
+  new Promise(resolve => {
+    if (Lbryio.authToken) {
+      resolve(Lbryio.authToken);
+    } else {
+      const store = global.store || (window.app ? window.app.store : null);
+      if (store) {
+        const state = store.getState();
+        const token = state.auth ? state.auth.authToken : null;
+        Lbryio.authToken = token;
+        resolve(token);
+      }
+
+      resolve(null);
+    }
+  });
+
+Lbryio.getCurrentUser = () => Lbryio.call('user', 'me');
+
+Lbryio.authenticate = () => {
+  if (!Lbryio.enabled) {
+    return new Promise(resolve => {
+      resolve({
+        id: 1,
+        language: 'en',
+        primary_email: 'disabled@lbry.io',
+        has_verified_email: true,
+        is_identity_verified: true,
+        is_reward_approved: false,
+      });
+    });
+  }
+
+  if (Lbryio.authenticationPromise === null) {
+    Lbryio.authenticationPromise = new Promise((resolve, reject) => {
+      Lbryio.getAuthToken()
+        .then(token => {
+          if (!token || token.length > 60) {
+            return false;
+          }
+
+          // check that token works
+          return Lbryio.getCurrentUser()
+            .then(() => true)
+            .catch(() => false);
+        })
+        .then(isTokenValid => {
+          if (isTokenValid) {
+            return reject;
+          }
+
+          return Lbry.status().then(status => {
+            const store = global.store || (window.app ? window.app.store : null);
```

Comment:
Do you need to check that `window` is an object so it doesn't break android?

Reply:
@seanyesmunt @akinwale if `window` isn't safe to use on Android then I'd propose the convention that it's unacceptable to use it in all shared JS code and instead must be passed in as a dependency. Littering code with `window && window.foo()` will hurt maintainability, readability, and introduce additional sources of error.

Reply:
To be clear, I'm not suggesting passing in `window`, specifically, but instead a generic term (such as `store` or whatever is appropriate) to be used in it's stead.

Reply:
Looks like `window` exists is actually already set to `global` React Native (https://github.com/facebook/react-native/blob/master/Libraries/Core/InitializeCore.js#L34-L36), so I can remove this.